### DISCUSSION TITLE
Search statistics

### DISCUSSION THREAD

**Discussion by FredrikEkstroem** [microsoft-search/pnp-modern-search#1462] [Q&A]
How does the built in search statistics (Microsoft Search/Configure search settings /_layouts/15/SiteAdmin.aspx) work with pnp modern search? I do get results for "Query volume trend", "Impressions by result type", "No result queries" and "Abandoned queries", but nothing for "Top Queries" and "Query rule usage". Is this something connected to how this project is set up or is the search service to blame? 

I use 4.4, SharePoint source and OOTB search field

**Comment by FranckyC** [comment]
Hi @FredrikEkstroem, there is no explicit set up in the modern search solution to integrate with search statistics (Microsoft Search). Maybe @wobba  can help on this as I didn't find any documentation about that. Maybe a special header/body property to pass along the query?

**Reply by wobba** [reply]
See https://techcommunity.microsoft.com/t5/microsoft-search-blog/we-re-improving-search-usage-reports-in-sharepoint-online-with/ba-p/1849502 for reports. And expect reports to track oob search, not API based.

Hopefully there will be an API we can use to feed API usecases into analytics. That said, you van build your own and track.

**Reply by FredrikEkstroem** [reply]
Tnx, I am using Google analytics for general search reports already, I just think the built in search statistics with abandoned queries and CTR are giving a very simple quick overview so it would be nice to hook into that. Hopefully the analytics will open up going forward...

**Comment by Wernbe** [comment]
Hi @wobba @FranckyC @FredrikEkstroem, We increasingly receive question about search statistics. Can we see what people search for? And use that to increase the knowledge base quality. Etc.. 

Are searches made from the modern webpart still exclude from the report? And if so, does somebody have a solution to track the modern webpart searches?

Kind regards!

**Reply by wobba** [reply]
The web parts are not logging anything. If you want logs you need to implement this yourself. I am discussing internally in Microsoft if and how queries via PnP could contribute to site/tenant analytics. The difficult part is how to record clicks and correlate with a query as there is o API today to record this.

**Reply by Wernbe** [reply]
Thanks you @wobba for your reply and for the great work you put in the webparts. Maybe you can give us some pointers where to put the logging in the code. Can we use an extension to write query to a sharepoint list?

**Reply by wobba** [reply]
You could create a webcomponent which you put around the <a> tags, which intercept the click and you log it somewhere. And add one around all the result to log the impression.

**Reply by Wernbe** [reply]
Thanks we will have a look at it.
